beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -240,19 +411,63 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
         );
     }
 
+    fn find_single_lookup_request(
+        &mut self,
+        target_id: Id,
+        stream_terminator: StreamTerminator,
+        response_type: ResponseType,
+    ) -> Option<(
+        bool,
+        &mut SingleBlockLookup<SINGLE_BLOCK_LOOKUP_MAX_ATTEMPTS, T>,
+    )> {
+        let lookup = self.single_block_lookups.iter_mut().find_map(|req| {
+            let id_opt = match response_type {
+                ResponseType::Block => req.id.block_request_id,
+                ResponseType::Blob => req.id.blob_request_id,
+            };
+            if let Some(lookup_id) = id_opt {
+                if lookup_id == target_id {
+                    // Only send for processing if we don't have parent requests that were triggered by
+                    // this block.
+                    let triggered_parent_request = self.parent_lookups.iter().any(|lookup| {
+                        lookup.chain_hash() == req.block_request_state.requested_block_root
```

Comment:
Does this need to check blob_state's requested blob_ids as well? I don't think it's required since parent requests make both block and blob requests simultaneously so checking just one should be sufficient. Just checking if that is the case.